### PR TITLE
docs: Added technologies chapter

### LINKED ISSUE #36: Start documentation with chapter 2

### PR DESCRIPTION
**Pull request by larsrickert**
Closes #36 .
Added chapter about used technologies for frontend and backend, which includes:
- TypeScript
- Sass
- Vue
- Strapi CMS

I didn't started the Heroku (deployment) section yet because we are still not 100% sure where the applications will be hosted.

### CODE REVIEW

File: docs/content/technologien.tex [hunk 1]
```diff
@@ -1,7 +1,136 @@
...
+  \centering
+  \caption[Verwendung einer Vue-Komponente]{Verwendung einer Vue \ac{SFC} \cites[vgl.][]{VueComponentBasics}[vgl.][]{VueSFC}}
+  \label{fig:vue-example-usage}
+\end{figure}
+
+\textbf{Sprach-Blöcke} \\
+In \acp{SFC} können standardmäßig die drei Sprach-Blöcke \lstinline{<template>} (HTML), \lstinline{<script>} (JavaScript) und \lstinline{<style>} (CSS) verwendet werden (siehe Abbildung \ref{fig:vue-example}), die den entsprechenden Code der Komponente beinhalten. Mit dem \lstinline{scoped}-Attribut des \lstinline{<style>}-Blocks kann das CSS für die Komponente gekapselt werden, d.h. es wird nur innerhalb der Komponente angewendet \cite[vgl.][]{VueSFC}.
+
+\textbf{TypeScript und Sass} \\
+Für die Entwicklung des Frontends sollen TypeScript (Version 4.4) und Sass (Version 1.43) verwendet werden, die bereits in Kapitel \ref{sec:typescript} und \ref{sec:sass} erläutert wurden. So soll sichergestellt werden, dass Fehler bereits während der Entwickler behoben werden können und ein robustes, wartbares Frontend entsteht.
```
Comment (philippabele) [inline docs/content/technologien.tex:97]: ... während der Entwicklung ...

File: docs/content/technologien.tex [hunk 2]
```diff
@@ -1,7 +1,136 @@
...
+  \centering
+  \caption[Beispiel einer Vue-Komponente mit Options API]{Beispiel einer Vue-Komponente mit Options API \cites[vgl.][]{VueIntroduction}[vgl.][]{VueSFC}}
+  \label{fig:vue-example-options-api}
+\end{figure}
+
+%
+% Vue - Reaktivität
+%
+\subsection{Reaktivität}
+Vue stellt mehrere Funktion zum Erstellen von reaktiven Variablen bereit. In Abbildung \ref{fig:vue-example} wurde bereits die Funktion \lstinline{ref} verwendet, die einen Wert reaktiv macht \cite[vgl.][]{VueApiReactiveCore}. Wird ein reaktiver Wert z.B. im \lstinline{<template>}-Block der Komponente verwendet, aktualisiert Vue automatisch den HTML-Code, wenn sich der Wert ändert \cite[vgl.][]{VueReactiveFundamentals}. Würde man also in Abbildung \ref{fig:vue-example} \glqq \lstinline{const count = ref(0)}\grqq{} durch \glqq \lstinline{const count = 0}\grqq{} ersetzen, würde im HTML-Code selbst bei einer Änderungen des Wertes immer \glqq \lstinline{You clicked me 0 times.}\grqq{} angezeigt werden, da \lstinline{count} dann nicht mehr reaktiv ist.
```
Comment (philippabele) [inline docs/content/technologien.tex:120]: ... mehrere Funktionen ...

File: docs/content/technologien.tex [hunk 3]
```diff
@@ -1,7 +1,136 @@
...
+
+Für weitere Funktionen bezüglich der Reaktivität sei für Interessierte auf \cite{VueApiReactiveCore} verwiesen. Für detaillierte Informationen über Vue empfiehlt sich die offizielle Dokumentation, die unter \cite{VueIntroduction} zu finden ist.
+
+%
+% Strapi CMS
+%
 \section{Strapi CMS}
+Strapi ist ein Open Source CMS, das in JavaScript geschrieben ist und vollständig anpassbar ist. Es soll die Entwicklung von APIs deutlich vereinfachen und kann selber gehostet werden. Dazu stellt es eine Benutzeroberfläche bereit, über die Inhaltstypen erstellt und verwaltet werden können. Anschließend sind diese durch REST oder GraphQL Endpunkte zugänglich \cite[vgl.][]{Strapi}.
```
Comment (philippabele) [inline docs/content/technologien.tex:128]: Was ist ein CMS ? Dem Leser erklären.

File: docs/content/technologien.tex [hunk 4]
```diff
@@ -1,7 +1,136 @@
...
+  \caption[Vue-Komponenten]{Strukturierung des Frontends durch Vue-Komponenten \cite{VueComponentBasics}}
+  \label{fig:vue-components}
+\end{figure}
+
+Vue-Komponenten erlauben die Unterteilung des Frontends in unabhängige, isolierte und wiederverwendbare Teile. So wird das Frontend in kleinere Komponenten gegliedert (siehe Abbildung \ref{fig:vue-components}), die dann die benötigte Logik kapseln und unabhängig entwickelt werden können \cite[vgl.][]{VueComponentBasics}.
+
+Vue bietet mehrere Möglichkeiten, um eine Komponente zu definieren. Im Rahmen dieser Studienarbeit werden sogenannte \aclu{SFC} (\acs{SFC}) verwendet (siehe Abbildung \ref{fig:vue-example}). Jede Komponente wird hierbei in einer Datei mit der Dateiendung \lstinline{.vue} definiert. Nachdem die Komponente in einer anderen Datei importiert wurde, kann sie wie ein herkömmliches HTML-Element verwendet werden (siehe Abbildung \ref{fig:vue-example-usage}) \cite[vgl.][]{VueComponentBasics}.
+
+\begin{figure}[!htb]
+  \includegraphics[width=0.9\textwidth]{images/vue-example.png}
```
Comment (philippabele) [inline docs/content/technologien.tex:80]: convert png into eps files using inkscape

**Review by philippabele — CHANGES_REQUESTED** [review]

File: docs/content/technologien.tex (hunk 4 above)
Comment (larsrickert) [reply docs/content/technologien.tex:80]: Latex cannot handle .eps files, they will be converted to pdf on compilation. Also what is the benefit of converting it?

File: docs/content/technologien.tex (hunk 3 above)
Comment (larsrickert) [reply docs/content/technologien.tex]: done :)

**Review by philippabele — APPROVED** [review]

**Pull request merged**